extensions/wikidata/tests/src/org/openrefine/wikidata/qa/scrutinizers/ConflictsWithScrutinizerTest.java

```diff
@@ -70,6 +95,33 @@ public void testNoValueSnak() {
 
         ItemUpdate updateA = new ItemUpdateBuilder(idA).addStatement(statement1).addStatement(statement2).build();
 
+        Stream<Statement> statementStream = conflictsWithParamaterStatementStream(entityIdValue, propertyParameterPID, conflictingPropertyValue, itemParameterPID, conflictingItemValue);
+
+        ConstraintFetcher fetcher = mock(ConstraintFetcher.class);
+        when(fetcher.getConstraintsByType(conflictsWithPid, "Q21502838")).thenReturn(statementStream);
+
+        setFetcher(fetcher);
+        scrutinize(updateA);
+        assertNoWarningRaised();
+    }
+
+    @Test
+    public void testNoStatement() {
```

Comment:
Really nice! That is exactly where the usefulness of mocking comes into play. Perhaps you could also add a test where multiple constraint definitions are present on the same property (we had found interesting examples of that on Wikidata).